jbehave-support-core/src/main/java/org/jbehavesupport/core/internal/web/WebScreenshotCreator.java

```diff
@@ -25,30 +26,60 @@
 @RequiredArgsConstructor
 public class WebScreenshotCreator {
 
-    public static final String SCREENSHOTS_KEY = "error_screenshots";
-    private static final String FILE_NAME_PATTERN = "FAILED_%s.png";
+    public enum Type {
+        FAILED(0), DEBUG(3), MANUAL(0), STEP(2), WAIT(1);
+        private final int hierarchy;
+
+        Type(int hierarchy){
+            this.hierarchy = hierarchy;
+        }
+
+        public int getHierarchy(){
+            return hierarchy;
+        }
+    }
+
+    public static final String FAILED_SCREENSHOTS_KEY = "error_screenshots";
+    public static final String STEP_SCREENSHOTS_KEY = "step_screenshots";
+    private static final String FILE_NAME_PATTERN = "%s_%s.png";
+
+    @Value("${web.screenshot.reporting.mode:MANUAL}")
+    private Type desiredType;
 
     @Value("${web.screenshot.directory:./target/reports}")
     private String screenshotDirectory;
 
     private final WebDriver driver;
     private final TestContext testContext;
 
-    public final void createScreenshot() {
-        try {
-            if (driver instanceof TakesScreenshot) {
-                log.info("Taking error screenshot will place it in {}", screenshotDirectory);
-                prepareDirectory();
-
-                File screenshot = ((TakesScreenshot)driver).getScreenshotAs(OutputType.FILE);
-                if (screenshot != null) {
-                    File destinationFile = getDestinationFile();
-                    FileUtils.copyFile(screenshot, destinationFile);
-                    storeInTestContext(destinationFile.getName());
+    public final void createScreenshot(Type screenShotType) {
+        if (desiredType.getHierarchy() >= screenShotType.getHierarchy()) {
+            try {
+                if (ExpectedConditions.alertIsPresent().apply(driver) != null) {
+                    log.info("Can't take screenshot (alert is present)");
+                } else if (driver instanceof TakesScreenshot) {
+                    log.info("Taking {} screenshot will place it in {}", screenShotType, screenshotDirectory);
+                    prepareDirectory();
+
+                    File screenshot = ((TakesScreenshot) driver).getScreenshotAs(OutputType.FILE);
+                    if (screenshot != null) {
```

Comment:
I think we should fail in case that screenshot is null and screenShotType is different than FAIL - for other cases we should always get screenshot and if not its probably an error.

Reply:
Done